x-pack/plugins/index_lifecycle_management/__jest__/client_integration/edit_policy/edit_policy.test.ts

```diff
@@ -672,66 +672,155 @@ describe('<EditPolicy />', () => {
+              testBed = await setup({ appServicesContext: { cloud: { isCloudEnabled: true } } });
+            });
+
+            const { component } = testBed;
+            component.update();
+          });
+          test('defaults searchable snapshot to true on cloud', async () => {
+            const { find, actions } = testBed;
+            await actions.cold.enable(true);
+            expect(
+              find('searchableSnapshotField-cold.searchableSnapshotToggle').props()['aria-checked']
+            ).toBe(true);
           });
-          httpRequestsMockHelpers.setListSnapshotRepos({ repositories: ['found-snapshots'] });
+        });
+        describe('existing policy', () => {
+          beforeEach(async () => {
+            httpRequestsMockHelpers.setLoadPolicies([getDefaultHotPhasePolicy('my_policy')]);
+            httpRequestsMockHelpers.setListNodes({
+              isUsingDeprecatedDataRoleConfig: false,
+              nodesByAttributes: { test: ['123'] },
+              nodesByRoles: { data_hot: ['123'] },
+            });
+            httpRequestsMockHelpers.setListSnapshotRepos({ repositories: ['found-snapshots'] });
+
+            await act(async () => {
+              testBed = await setup({ appServicesContext: { cloud: { isCloudEnabled: true } } });
+            });
+
+            const { component } = testBed;
+            component.update();
+          });
+
+          test('should show off, custom and data role options on cloud with data roles', async () => {
+            const { actions, find } = testBed;
 
-          await act(async () => {
-            testBed = await setup({ appServicesContext: { cloud: { isCloudEnabled: true } } });
+            await actions.warm.enable(true);
+            actions.warm.showDataAllocationOptions();
+            expect(find('defaultDataAllocationOption').exists()).toBeTruthy();
+            expect(find('customDataAllocationOption').exists()).toBeTruthy();
+            expect(find('noneDataAllocationOption').exists()).toBeTruthy();
+            // We should not be showing the call-to-action for users to activate data tiers in cloud
+            expect(find('cloudDataTierCallout').exists()).toBeFalsy();
           });
 
-          const { component } = testBed;
-          component.update();
-        });
-        test('correctly sets snapshot repository default to "found-snapshots"', async () => {
-          const { actions } = testBed;
-          await actions.cold.enable(true);
-          await actions.cold.toggleSearchableSnapshot(true);
-          await actions.savePolicy();
-          const latestRequest = server.requests[server.requests.length - 1];
-          const request = JSON.parse(JSON.parse(latestRequest.requestBody).body);
-          expect(request.phases.cold.actions.searchable_snapshot.snapshot_repository).toEqual(
-            'found-snapshots'
-          );
+          test('should show cloud notice when cold tier nodes do not exist', async () => {
```

Comment:
Should we have complementary assertions in the "on cloud using legacy data role config pre v8" tests?